### PR TITLE
Stricter rules for custom user agent

### PR DESCRIPTION
**Pull request by pinheadmz**
Closes https://github.com/handshake-org/hnsd/issues/96

Blocked by #101 and #102 and #99 😬 

```
--> ./hnsd -a bad/agent
chain (0): chain is fully synced
addrman: added addr: 127.0.0.1:46806
failed adding user agent
failed initializing daemon: EFAILURE

--> ./hnsd -a good_agent
...
[info] (net) Received version (127.0.0.1:60772): version=1 height=0 services=0 agent=/hnsd:1.0.0/good_agent/
```

### CODE REVIEW

File: integration/test/config-test.js [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+
+  before(async () => {
+    await util.open();
+  });
+
+  after(async () => {
+    await util.close();
+  });
+
+  describe.only('--agent', function () {
```
Comment (rithvikvibhu) [inline integration/test/config-test.js:25]: `.only` to be removed? All tests pass without this.

File: integration/test/config-test.js [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+    it('should have default user agent', async () => {
+      await util.generate(1);
+      await util.waitForSync();
+
+      const peer = util.node.pool.peers.head();
+      assert.strictEqual(peer.agent, '/hnsd:1.0.0/');
```
Comment (rithvikvibhu) [inline integration/test/config-test.js:31]: Does the test need to be updated with every version bump? Could pick from package.json (which I think was the idea for adding version there)

Edit: yeah just saw version used below

**Review by rithvikvibhu — COMMENTED** [review]
Agent commit looks good, tests run fine! Left some comments.

File: integration/test/config-test.js (hunk 1 above)
Comment (pinheadmz) [reply integration/test/config-test.js:25]: yep thanks

File: integration/test/config-test.js (hunk 2 above)
Comment (pinheadmz) [reply integration/test/config-test.js:31]: yep! fixing...

**Pull request closed without merging**